### PR TITLE
Fixings for gitlab ci

### PR DESCRIPTION
**Pull request by ailisp**
- Updating context and config.json to most recent
- remove main.wat, thanks @bowenwang1996 for the note and I put my point below.
- fix test script so
  - it use up to date error message check
  - do not exit, instead run all tests and gives a report like other unit test framework did:

image

Update 9/5/19:
All test passed if use the near-shell way building. Original way build generates a different wasm that fails in above screenshot. 
image

### CODE REVIEW

**Comment by bowenwang1996** [comment]
Why is `main.wat` removed? Sometimes it's helpful to look at the wasm file and see what changed.

**Comment by ailisp** [comment]
> Why is `main.wat` removed? Sometimes it's helpful to look at the wasm file and see what changed.

generated wat file can be useful but it's still generated that should not be included in src and it's exclude by `.gitignore` so to me it's "added by mistake". Also, it is hard to tell which version of src that current wat is built from so it's better to not include this wat. Instead, for your suggest workflow, let's do:
1. checkout to old commit hash. do a build to ensure you get the old working wat, mv it to `main.wat.old`
2. checkout to new commit hash. build.
3. run diff locally

**Comment by ailisp** [comment]
@janedegtiareva need some input on how to run tests on standalone vm on ci. Yesterday I said install nodejs/npm in a nearprotocol/nearcore image, but it's problematic because nearcore image 1) only build near executable (no near-vm-standalone); 2) nearcore source code deleted in last step in dockerfile.

Some alternative approach:
1. from node:12 image install rustup, clone and build near-vm-standalone. I'm not sure about this choice:
    a. use a hard coded git hash of nearcore in gitlabci.yml; 
    b. or always clone the most recent master or staging.
    c. or a+b to prove it works with specific nearcore versions + "nightly" nearcore?
2. use a git submodule here
3. release near-vm-standalone executable to github release. download and test with it in node:12 image. Proper place to set this will be ci in nearcore.

What do you think?

**Comment by ailisp** [comment]
Gitlab ci is ready now!
image

File: tests/build-near-vm-runner-standalone.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    git clone https://github.com/nearprotocol/nearcore
+    cd nearcore
+fi
+
+if [[ ! -e $HOME/.cargo ]]; then
+curl https://sh.rustup.rs -sSf | \
+     sh -s -- -y --no-modify-path --default-toolchain $RUST_VERSION
+fi
+
+$HOME/.cargo/bin/cargo build -p near-vm-runner-standalone
```
Comment (janedegtiareva) [inline tests/build-near-vm-runner-standalone.sh]: you can also do cargo install near-vm-runner-standalone and then you don't need to depend on nearcore. that is going to be faster, but will not catch problems on nearcore changes.

**Comment by janedegtiareva** [comment]
Another point to consider: we potentially also want to run this test on a real node. This will require a nearlib dependency since we need to deploy a contract, so this project is not the best place to do it. We should do this in a separate "integration test" repo where we can get all the latest dependencies.

File: tests/build-near-vm-runner-standalone.sh (hunk 1 above)
Comment (ailisp) [reply tests/build-near-vm-runner-standalone.sh]: That's a good point. This script was meant to be working both locally and on CI so it has some compromise.

**Comment by ailisp** [comment]
> Another point to consider: we potentially also want to run this test on a real node. This will require a nearlib dependency since we need to deploy a contract, so this project is not the best place to do it. We should do this in a separate "integration test" repo where we can get all the latest dependencies.

Good point. do we already have a repo (for example does near-e2e-test mean to do that)

**Pull request merged**